### PR TITLE
Refactor SMQTK search

### PR DESCRIPTION
**Pull request by danlamanna**
@jeffbaumes 

Primarily, this removes the Mongo db mapping that was required (and undocumented). It also generalizes the Solr chunked request utility and adds an event so I was able to move some of the uppercase logic out of ImageSpace proper and into the weapons plugin.

### CODE REVIEW

File: imagespace_weapons/server/__init__.py [hunk 1]
```diff
@@ -24,7 +25,18 @@ def add_maintype_to_qparams(event):
     event.addResponse(event.info)
 
 
+def uppercase_basename_for_resourcenames(event):
```
Comment (jeffbaumes) [inline imagespace_weapons/server/__init__.py:27]: Would be good to document in a comment why this is needed (as hacky as it may be).

**Comment by danlamanna** [comment]
@jeffbaumes Can you review this? Just ensuring basic similarity searches still work.

**Comment by jeffbaumes** [comment]
LGTM

File: imagespace_smqtk/server/__init__.py [hunk 2]
```diff
@@ -22,8 +22,7 @@
 
 
 def load(info):
-    required_env_vars = ('IMAGE_SPACE_SMQTK_SIMILARITY_SEARCH',
-                         'IMAGE_SPACE_SMQTK_MAP_COLLECTION')
+    required_env_vars = ('IMAGE_SPACE_SMQTK_SIMILARITY_SEARCH',)
```
Comment (chrismattmann) [inline imagespace_smqtk/server/__init__.py:25]: can we make sure to update the README here for this?

**Comment by chrismattmann** [comment]
LGTM but needs README update.

**Comment by danlamanna** [comment]
Added a readme for the SMQTK plugin documenting the new environment variable.

Thanks all.

**Comment by chrismattmann** [comment]
We also need to update the main ImageSpace README.md and explain how to configure and use these plugins. Thanks @danlamanna

**Pull request merged**